api_inventory_test.go

```diff
@@ -1079,3 +1079,54 @@ func TestApiGetDeviceGroup(t *testing.T) {
 		runTestRequest(t, apih, tc.inReq, tc.JSONResponseParams)
 	}
 }
+
+func TestApiDeleteDevice(t *testing.T) {
+	rest.ErrorFieldName = "error"
+
+	tcases := map[string]struct {
+		utils.JSONResponseParams
+
+		inReq        *http.Request
+		inDevId      DeviceID
+		inventoryErr error
+	}{
+		"no device": {
+			inDevId:      DeviceID("1"),
+			inReq:        test.MakeSimpleRequest("DELETE", "http://1.2.3.4/api/0.1.0/devices/1", nil),
+			inventoryErr: ErrDevNotFound,
+			JSONResponseParams: utils.JSONResponseParams{
+				OutputStatus:     http.StatusNotFound,
+				OutputBodyObject: RestError(ErrDevNotFound.Error()),
+			},
+		},
+		"some device": {
+			inDevId: DeviceID("2"),
+			inReq:   test.MakeSimpleRequest("DELETE", "http://1.2.3.4/api/0.1.0/devices/2", nil),
+			JSONResponseParams: utils.JSONResponseParams{
+				OutputStatus: http.StatusNoContent,
+			},
+		},
+		"error": {
+			inDevId: DeviceID("3"),
+			inReq:   test.MakeSimpleRequest("DELETE", "http://1.2.3.4/api/0.1.0/devices/3", nil),
+			JSONResponseParams: utils.JSONResponseParams{
+				OutputStatus:     http.StatusInternalServerError,
+				OutputBodyObject: RestError("internal error"),
+			},
+			inventoryErr: errors.New("inventory internal error"),
+		},
+	}
+
+	for name, tc := range tcases {
+		t.Logf("test case: %s", name)
```

Comment:
Just a reminder, you'll have to use a pattern like this: https://github.com/mendersoftware/deviceauth/blob/master/api_devauth_test.go#L472 to be able to use `t.Parallel()` with `for` loop. Otherwise `tc` is just a loop variable and all test cases will run with the same (last) `tc`